java/system-test/src/main/java/sleeper/systemtest/cdk/SystemTestApp.java

```diff
@@ -47,9 +54,20 @@ public SystemTestApp(App app, String id, StackProps props, SystemTestProperties
 
     @Override
     public void create() {
-        super.create();
         SystemTestProperties systemTestProperties = getInstanceProperties();
-
+        List<String> optionalStacks = systemTestProperties.getList(OPTIONAL_STACKS);
+        List<IBucket> systemTestIngestBuckets = new ArrayList<>();
+        if (INGEST_STACK_NAMES.stream().anyMatch(optionalStacks::contains)) {
+            String systemTestIngestBucketName = String.join("-", "sleeper", systemTestProperties.get(ID),
+                    "system", "test", "ingest").toLowerCase(Locale.ROOT);
+            systemTestProperties.set(INGEST_SOURCE_BUCKET, systemTestIngestBucketName);
+            SystemTestIngestBucketStack ingestBucketStack = new SystemTestIngestBucketStack(this,
+                    "SystemTestIngestBucket", systemTestIngestBucketName);
+            systemTestProperties.getTags()
+                    .forEach((key, value) -> Tags.of(ingestBucketStack).add(key, value));
+            systemTestIngestBuckets.add(ingestBucketStack.getIngestBucket());
```

Comment:
It seem like most of this could be in SystemTestIngestBucketStack, and SystemTestStack could reference the buckets the same way as the main stack does, ie. with IngestStack.addIngestSourceBucketReferences.